akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -579,7 +581,10 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
         val fastFuture = FastFuture.successful(gateway)
         hostPoolCache.put(setup, fastFuture) // optimize subsequent gateway accesses
         gatewayPromise.success(gateway) // satisfy everyone who got a hold of our promise while we were starting up
-        whenShuttingDown.future.onComplete(_ ⇒ hostPoolCache.remove(setup, fastFuture))(fm.executionContext)
+        whenShuttingDown.future.onComplete { _ ⇒
+          hostPoolCache.remove(setup, fastFuture)
```

Comment:
Yes, if this isn't done before the shutdown-future inside `PoolGateway` completes, it might be able to fetch itself from the `hostPoolCache`, which causes the issue.